### PR TITLE
fix: Use passlib for totp_secret

### PR DESCRIPTION
**Pull request by jwag956**
onetimepass seems to be unmaintained - and more importantly totp_secret was being
stored in the DB in the clear as well as in the session cookie in the clear.
That is a big security hole.
passlib has an easy way to encrypt these.

Also:
- updated docs
- make sure that on logout, all two factor session elements are cleared out
- Using two-factor now requires a TWO_FACTOR_SECRET as defined by passlib

This change requires the setting of TWO_FACTOR_SECRET, and installing the cryptography
package.

closes: #89

### CODE REVIEW

File: tests/test_two_factor.py [hunk 1]
```diff
@@ -213,14 +229,22 @@ def test_two_factor_flag(app, client):
     response = client.post('/login', data=json_data,
                            headers={'Content-Type': 'application/json'},
                            follow_redirects=True)
-    totp_secret = u'RCTE75AP2GWLZIFR'
-    code = str(onetimepass.get_totp(totp_secret))
+    session = get_session(response)
+
+    # This shows how dangerous it is to have even an encrypted totp_secret
```
Comment (baurt) [inline tests/test_two_factor.py]: I am unsure of any other way to provide the verification of the second layer.  I'm sure there's a way, I just don't know what it is.

File: docs/features.rst [hunk 2]
```diff
@@ -79,6 +79,7 @@ known limitations:
     * Limited and incomplete JSON support
     * Incomplete i18n support
     * Not enough documentation to use w/o looking at code
+    * (low) While confirming 2FA - encrypted TOTP is sent in cookie
```
Comment (baurt) [inline docs/features.rst]: low?  Not sure what that means?

File: tests/test_two_factor.py (hunk 1 above)
Comment (baurt) [reply tests/test_two_factor.py]: Maybe just store it on the user model and call it from there?

Comment (jwag956) [reply tests/test_two_factor.py]: That was my thought as well - but need to look into it more.

**Comment by jwag956** [comment]
> Looks good. I don't think pyqrcode is maintained anymore either. The package https://github.com/lincolnloop/python-qrcode looks to be up to date. This is probably another pull though.
> 
> I say merge it.

Interesting - the passlib tutorial actually still references pyqrcode....

File: docs/features.rst (hunk 2 above)
Comment (jwag956) [reply docs/features.rst]: removed wording.

**Comment by baurt** [comment]
> > Looks good. I don't think pyqrcode is maintained anymore either. The package https://github.com/lincolnloop/python-qrcode looks to be up to date. This is probably another pull though.
> > I say merge it.
> 
> Interesting - the passlib tutorial actually still references pyqrcode....

The last release was in June 2016.  It works for what it's used for.  I don't think that the QR codes are necessarily an issue though since they only appear during setup.

**Pull request merged**